lib/FusionInventory/Agent/Tools/License.pm

```diff
@@ -72,7 +73,54 @@ sub getAdobeLicenses {
             NAME       => $key,
             FULLNAME   => $data{$key}{ALM_LicInfo_EpicAppName},
             KEY        => _decodeAdobeKey($data{$key}{SN}),
-            COMPONENTS => join('/', @{$data{$key}{with}})
+            COMPONENTS => join('/', sort @{$data{$key}{with}})
+        }
+    }
+
+    return @licenses;
+}
+
+sub getAdobeLicensesWithoutSqlite {
+    my ($fileAdobe) = @_;
+    my $contentFileAdobe;
+
+    my @licenses;
+
+    open(my $fh, '<', $fileAdobe) or die "cannot open file $fileAdobe";
```

Comment:
You should use `getAllLines` from `FusionInventory::Agent::Tools` module here to set $contentFileAdobe with something like `my $contentFileAdobe = getAllLines($fileAdobe) or return;`